src/lib/sql/sql_pipeline.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <string>
+
+#include "SQLParserResult.h"
+#include "concurrency/transaction_context.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "sql/sql_query_plan.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+/**
+ * This is the unified interface to handle SQL queries and related operations.
+ *
+ * The basic idea of the SQLPipeline is that represents the flow from the basic SQL string to the result table with all
+ * intermediate steps. These intermediate steps call the previous step that is required. The intermediate results are
+ * all cached so calling a method twice will return the already existing value.
+ *
+ * The SQLPipeline holds all results and only hands them out as const references. If the SQLPipeline goes out of scope
+ * while the results are still needed, the results are either nullptr or undefined.
```

Comment:
nullptr? If the Pipeline goes out of scope, all references handed out will be invalid, except  for the result table (since its a shared_ptr)